DataBase/CSVDataBase.py

```diff
@@ -17,4 +22,110 @@ def update_data(self):
+        urllib.request.urlretrieve(url, filepath)
+        return
+
+    def _get_wb_data(file_path,data_name):
+        #http://api.worldbank.org/v2/en/indicator/SP.POP.TOTL?downloadformat=csv
+        df = pd.read_csv(file_path,skiprows=4)
+        series = df.set_index('Country Name')['2018']
+        series.name = data_name
+        
+        #fix the mismatched names in the world bank data
+        with open('name_mapping.json') as json_file:
+            name_mapping = json.load(json_file)
+        
+        data = series.reset_index()
+        data['Country Name'].replace(name_mapping,inplace=True)
+        data.set_index('Country Name',inplace=True)
+        
+        return data
+
+    def _join_wb_data():
+        #get pop density
+        #http://api.worldbank.org/v2/en/indicator/EN.POP.DNST?downloadformat=csv
+        pop_density = get_wb_data('API_EN.POP.DNST_DS2_en_csv_v2_936296.csv','pop_density')
+        #get population
+        #http://api.worldbank.org/v2/en/indicator/SP.POP.TOTL?downloadformat=csv
+        pop = get_wb_data('API_SP.POP.TOTL_DS2_en_csv_v2_936048.csv','pop')
+        
+        wb_data = pd.concat([pop_density,pop],axis=1)
+        return wb_data
+
+    def _join_data_sources():
+        df = scraper.get_latest_data()
+        df = df.join(join_wb_data())
+        return df
+
+    def _generate_features(df):
+        df['deaths_per_million'] = df['deaths'] / df['pop'] * 1000000
+        df['cases_per_million'] = df['cases'] / df['pop'] * 1000000
+        #calculate additional metrics
+        df['death_rate_%'] = df['deaths'] / df['cases'] * 100
+        return df
+        
+    def _import_static_data():
+        df = join_data_sources()
+        df = generate_features(df)
+        return df
+
+    def _get_csse_time_series_deaths():
+        df = pd.read_csv("https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_deaths_global.csv")
+        meta_cols = ['Province/State','Country/Region','Lat','Long']
+        time_series = df.drop(meta_cols,axis=1)
+        #get a unique country name to use as id
+        df['Province/State'] = df['Province/State'].fillna("")
+        df['country'] = df['Country/Region'] + " " + df['Province/State']
+        df['country'] = df['country'].str.rstrip()
+        #pivot the dataframe
+        time_series = time_series.T
+        time_series.columns = df['country']
+        #melt countries and deaths into single column
+        df_melted = pd.melt(time_series.reset_index(), id_vars=["index"])
+        df_melted.columns = ['date','country','deaths']
+        df_melted['date'] = pd.to_datetime(df_melted['date'])
+        
+        return df_melted
+
+    def _get_apple_movement_indices(movement_type='walking'):
+        """
+        get latest time series data from apple on population movement by country
+            - movement_type <str> enum "walking"|"driving|"transit"
+        """
+        try:
```

Comment:
this is the code from before i merged a change yesterday, can you update it with the current version of this method from master?

Reply:
pulled and merged most recent code